library/libwaku.nim

```diff
@@ -43,14 +45,14 @@ const RET_MISSING_CALLBACK: cint = 2
 proc relayEventCallback(ctx: ptr Context): WakuRelayHandler =
   return proc (pubsubTopic: PubsubTopic, msg: WakuMessage): Future[system.void]{.async.} =
     # Callback that hadles the Waku Relay events. i.e. messages or errors.
-    if not isNil(ctx[].eventCallback):
+    if not isNil(ctx[].eventCallback) and not isNil(ctx[].eventUserData):
```

Comment:
Given that we now have two conditions that should be met, I wonder if we could split the validation in two, and apply some kind of  "return early pattern". The following article explains the idea much better:
https://medium.com/swlh/return-early-pattern-3d18a41bba8

lmk what do you think :)

Reply:
Sounds good! I removed the validation from the `if` and executed those following the 'return early' pattern as described in the article you shared.
https://github.com/waku-org/nwaku/pull/2435/commits/a4f7b88bacdbd30fe7a76654d4e3a732ddeb1f66